integration/spark/src/main/scala/org/apache/spark/sql/events/MergeIndexEventListener.scala

```diff
@@ -43,6 +43,11 @@ class MergeIndexEventListener extends OperationEventListener with Logging {
   override def onEvent(event: Event, operationContext: OperationContext): Unit = {
     event match {
       case preStatusUpdateEvent: LoadTablePreStatusUpdateEvent =>
+        // skip merge index in case of insert stage flow
+        if (null != operationContext.getProperty("isInsertStage") &&
```

Comment:
Define "isInsertStage" as a constant will be better.

Reply:
done